lib/ProductOpener/Store.pm

```diff
@@ -68,43 +68,44 @@ sub unac_string_perl($) {
 }
 
 # Tags in European characters (iso-8859-1 / Latin-1 / Windows-1252) are canonicalized:
-# 1. deaccent: é -> è, + German umlauts: ä -> ae
-# 2. lowercase
+# 1. lowercase
+# 2. deaccent: é -> è, + German umlauts: ä -> ae if $unaccent is 1 OR $lc is 'fr'
 # 3. turn ascii characters that are not letters / numbers to -
 # 4. keep other UTF-8 characters (e.g. Chinese, Japanese, Korean, Arabic, Hebrew etc.) untouched
 # 5. remove leading and trailing -, turn multiple - to -
 
-sub get_fileid($) {
+sub get_fileid {
```

Comment:
I would prefer to keep the ($$$) to indicate that the arguments are mandatory, so that we get errors at compile time. Otherwise we might forget one argument by error.

Maybe we could change the name of the function (I named it at the very beginning of the project when it was only used for filenames, but now it's mostly used for tag names). How about something like this?

sub get_string_id_for_language($$) {

my $string = shift;
my $string_lc = shift; # undef to get the default normalization for strings that are not in a specific language like user names

Then accenting / deaccenting is decided from the $string_lc parameter.

Reply:
https://stackoverflow.com/a/297265/11963 claims that
> Perl's prototypes have a singular and very different purpose:
> Prototypes allow you to define functions that behave like built-in functions.

Because of that, I'm a bit torn about keeping them. I was actually considering creating an issue about removing them altogether, but that's off-topic for this PR. :)

I'll have a look at renaming the sub and will integrate it into this PR if it goes well.

Reply:
> I'll have a look at renaming the sub and will integrate it into this PR if it goes well.

Doesn't work without creating a duplicate or misusing the `$string_lc` parameter, because we need forced unaccenting for actual file IO  (ie. reset_password.pl finds a `"$data_root/users/id.sto"` file with that).

Reply:
For things that in fact don't have a language (like usernames), we can just pass undef as the language, and we apply the current default unaccenting. The decision to unaccent a specific language can be put in the configuration file (and left to the people who administer a particular instance), instead of being set in the code in the calls to the normalization function.